### ISSUE TITLE
many c-ares mock tests are failing on one of my two machines [failing on Pop!_os and Alpine,  passing on osx]

### ISSUE THREAD

**Issue by tklebanoff** [c-ares/c-ares#261]
Hello c-ares developers!

I am working with the c-ares build system and am noticing that a great many tests fail when I build on Pop!_os.  It looks like these stem from the same issue, but I am new to the project and am unsure where to start or what to check.  My hunch is that it is actually just something simple that I am missing.  I havn't been able to find the answer after studying the build system for a few hours -- I was hoping that somebody here could help me! Thanks in advance for any help!

I ran all of the tests successfully on OSX a few moments ago.  I tried running the tests inside of a docker container, and they failed there with the same errors as outside of the container on my linux machine.

Is there anything in the system configuration on linux that I might be overlooking? I have linked to a pastebin report of the errors I see

https://pastebin.com/raw/PdijG3zF

The command I am running from the build directory is:
cmake .. -DCARES_BUILD_TESTS:BOOL=ON && make -j 4 && ./bin/arestest -v

**Comment by bradh352** [comment]
@daviddrysdale Did you write these tests that are failing?  I don't know enough about the MockChannel tests to understand why they'd fail (and don't have the time at the moment to really dig into it).

As far as I can tell, its all fully simulated responses being validated, nothing live or environment specific, so I don't know why it would fail.

**Comment by daviddrysdale** [comment]
I did write them, but it was a long time ago!
 
@tklebanoff, is the transcript from a run with `-v` turned on?  (Your command includes it but the transcript doesn't have [some output](https://github.com/c-ares/c-ares/blob/master/test/ares-test.cc#L213-218) I'd expect to see.)

Also, does Pop!_os have any restrictions on UDP ports that are available to userspace programs?   (If so, see if changing the default `-p 5300` option helps).  If the mock server was failing to listen for packets, that would explain the failures.

**Comment by bradh352** [comment]
closing, no reply

**Comment by klebz** [comment]
Hello! I am sorry for such a delay in my response! (this is @tklebanoff, though from a more current account)

Long story short, in 2019 I ended up off grid in a jungle on the other side of the planet for a stretch --  When I came back, I had most of my computer equipment (including the PopOS system) stowed in trunks loaded into the back of my truck.

I drove coast to coast back to the east where I grew up.

At that point I was working off a non PopOS laptop and lost track of this thread while the pandemic was going on.

I do want to thank you both for the time you each took to reply and also indicate that although it might be a minor thing, I have not forgotten to respond here!

@daviddrysdale I think, if I remember correctly, the problem I was encountering matches your idea about the restricted UDP ports for userspace programs.  The next time I get back to this part of the systems, I will be sure to take a look and post my results here.

I am not currently working on PopOS, but that might change soon.

I don't have anything further to add to this case at the moment. I don't think we need to reopen it.

I just wanted to make sure to thank you both for your help and indicate that I had not forgotten it!!

Cheers!

-TK
